### PR TITLE
refactor(elb): fix elb tf features issues

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  fix elb tf features issues
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  fix elb tf features issues
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/elb/' TESTARGS='-run  TestAccElbFlavorsDataSource_basic' 
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/elb/ -v -run  TestAccElbFlavorsDataSource_basic -timeout 360m -parallel 4 
=== RUN   TestAccElbFlavorsDataSource_basic 
=== PAUSE TestAccElbFlavorsDataSource_basic 
=== CONT  TestAccElbFlavorsDataSource_basic
--- PASS: TestAccElbFlavorsDataSource_basic (8.00s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/elb       8.052s
```

### CODE REVIEW

File: huaweicloud/services/elb/data_source_huaweicloud_elb_flavors.go [hunk 1]
```diff
@@ -101,27 +104,27 @@ func dataSourceElbFlavorsV3Read(d *schema.ResourceData, meta interface{}) error
 
 	pages, err := flavors.List(elbClient, listOpts).AllPages()
 	if err != nil {
-		return err
+		return diag.Errorf("%s", err)
```
Comment (ShiChangkuo) [inline huaweicloud/services/elb/data_source_huaweicloud_elb_flavors.go]:
```suggestion
		return diag.FromErr(err)
```

File: huaweicloud/services/elb/data_source_huaweicloud_elb_flavors.go [hunk 2]
```diff
@@ -138,29 +141,32 @@ func dataSourceElbFlavorsV3Read(d *schema.ResourceData, meta interface{}) error
...
-	d.Set("region", config.GetRegion(d))
 
-	return nil
+	var mErr *multierror.Error
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", config.GetRegion(d)),
+		d.Set("ids", ids),
+		d.Set("flavors", flavorInfos),
+	)
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/data_source_huaweicloud_elb_flavors.go]:
```suggestion
	mErr := multierror.Append(
		mErr,
		d.Set("region", config.GetRegion(d)),
		d.Set("ids", ids),
		d.Set("flavors", flavorInfos),
	)
```

**Review by chengxiangdong — COMMENTED** [review]
建议优化一下 `diag.Errorf` 输出的日志，推荐使用首字母小写，因为 Terraform 会拼接。
需要注意，如果是在 `log.Print` 直接输出到文件的日志，首字母应当大写。

File: huaweicloud/services/elb/data_source_huaweicloud_elb_flavors.go [hunk 3]
```diff
@@ -87,11 +90,11 @@ func DataSourceElbFlavorsV3() *schema.Resource {
 	}
 }
 
-func dataSourceElbFlavorsV3Read(d *schema.ResourceData, meta interface{}) error {
+func dataSourceElbFlavorsV3Read(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*config.Config)
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/data_source_huaweicloud_elb_flavors.go]: `config` is same with the package name, it is recommended to rename it

File: huaweicloud/services/elb/data_source_huaweicloud_elb_flavors.go [hunk 4]
```diff
@@ -87,11 +90,11 @@ func DataSourceElbFlavorsV3() *schema.Resource {
...
 	elbClient, err := config.ElbV3Client(config.GetRegion(d))
 	if err != nil {
-		return fmtp.Errorf("Error creating HuaweiCloud elb v3 client: %s", err)
+		return diag.Errorf("Error creating ELB client: %s", err)
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/data_source_huaweicloud_elb_flavors.go]:
```suggestion
		return diag.Errorf("error creating ELB client: %s", err)
```

File: huaweicloud/services/elb/data_source_huaweicloud_elb_flavors.go [hunk 5]
```diff
@@ -101,27 +104,27 @@ func dataSourceElbFlavorsV3Read(d *schema.ResourceData, meta interface{}) error
...
 	pages, err := flavors.List(elbClient, listOpts).AllPages()
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 
 	allFlavors, err := flavors.ExtractFlavors(pages)
 	if err != nil {
-		return fmtp.Errorf("Unable to retrieve flavors: %s", err)
+		return diag.Errorf("Unable to retrieve flavors: %s", err)
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/data_source_huaweicloud_elb_flavors.go]:
```suggestion
		return diag.Errorf("unable to retrieve flavors: %s", err)
```

File: huaweicloud/services/elb/data_source_huaweicloud_elb_flavors.go [hunk 6]
```diff
@@ -138,29 +141,32 @@ func dataSourceElbFlavorsV3Read(d *schema.ResourceData, meta interface{}) error
...
-			"bandwidth":       int(flavor.Info.Bandwidth / 1000),
+			"bandwidth":       flavor.Info.Bandwidth / 1000,
 		}
-		s = append(s, mapping)
+		flavorInfos = append(flavorInfos, flavorInfo)
 	}
 
 	if len(ids) < 1 {
-		return fmtp.Errorf("Your query returned no results. " +
+		return diag.Errorf("Your query returned no results. " +
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/data_source_huaweicloud_elb_flavors.go]:
```suggestion
		return diag.Errorf("your query returned no results. " +
```

File: huaweicloud/services/elb/data_source_huaweicloud_elb_flavors.go (hunk 1 above)
Comment (houpeng80) [reply huaweicloud/services/elb/data_source_huaweicloud_elb_flavors.go]: done

Comment (houpeng80) [reply huaweicloud/services/elb/data_source_huaweicloud_elb_flavors.go]: done

File: huaweicloud/services/elb/data_source_huaweicloud_elb_flavors.go (hunk 2 above)
Comment (houpeng80) [reply huaweicloud/services/elb/data_source_huaweicloud_elb_flavors.go]: done

File: huaweicloud/services/elb/data_source_huaweicloud_elb_flavors.go (hunk 3 above)
Comment (houpeng80) [reply huaweicloud/services/elb/data_source_huaweicloud_elb_flavors.go]: done

File: huaweicloud/services/elb/data_source_huaweicloud_elb_flavors.go (hunk 4 above)
Comment (houpeng80) [reply huaweicloud/services/elb/data_source_huaweicloud_elb_flavors.go]: done

File: huaweicloud/services/elb/data_source_huaweicloud_elb_flavors.go (hunk 5 above)
Comment (houpeng80) [reply huaweicloud/services/elb/data_source_huaweicloud_elb_flavors.go]: done

File: huaweicloud/services/elb/data_source_huaweicloud_elb_flavors.go (hunk 6 above)
Comment (houpeng80) [reply huaweicloud/services/elb/data_source_huaweicloud_elb_flavors.go]: done

**Pull request merged**